lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -372,8 +372,9 @@ export class PublicClientApplication {
             return await this.authModule.getValidToken(silentRequest);
         } catch (e) {
             const isServerError = e instanceof ServerError;
+            const isInteractionRequiredError = e instanceof InteractionRequiredAuthError;
             const isInvalidGrantError = (e.errorCode === BrowserConstants.INVALID_GRANT_ERROR);
-            if (isServerError && isInvalidGrantError) {
+            if (isServerError && isInvalidGrantError && !isInteractionRequiredError) {
```

Comment:
I'm thinking we should make this just `!isInteractionRequiredError`. Do we only want to run through the whole flow of getting a code and exchanging it for a token if the access token is expired? Or anytime it fails for a reason other than needing interaction?

Reply:
The silent request that proceeds this if statement is specifically to handle the expired RT scenario (which manifests as the invalid grant error).

Reply:
Right but is that the only scenario we want to cover with this? An example I ran into when testing was when there was no Auth Code or RT available. It would fail to get a token above, but then would not enter this if block and the silent request would fail with a ClientAuthError. Is that desired?

Reply:
I believe so, @pkanher617 can confirm.

Reply:
The scenario I mentioned above will be handled in a different place in a different PR